### ISSUE TITLE
Restore link to 'Your Profile' in menu, but make linkable URL obvious

### ISSUE THREAD

**Issue by tog22** [tog22/eahub#11]
@cdc482 pointed this out first - we had a link to http://effectivealtruismhub.com/user called 'Your Profile' in the menu. Nicely, this takes you to the login/registration page if you're not logged in, and to your Profile if you are. The only problem is that in the latter case it still shows http://effectivealtruismhub.com/user in the address bar rather than redirecting to e.g. http://effectivealtruismhub.com/user/peter-singer - the only address that'd work as a pastable link for _other_ people.

This has become more problematic with the imminent resolution of https://github.com/tog22/eaforum/issues/1 thanks to @arichard4 . So I've removed the link at http://effectivealtruismhub.com/admin/structure/menu/manage/main-menu for now. This will however make it hard for all the people about to remind themselves of their EA Profile URLs due to  tog22/eaforum/issues/1 . So this task would be restore the link to 'Your Profile' in the menu, but make linkable URL obvious.

The neatest approach would be to redirect logged in users going to http://effectivealtruismhub.com/user to their full profile URL, affecting the address bar. This would almost certainly involve using a Drupal hook in template.php in the theme folder (or using it in one of our custom modules). However it'd be fine, and probably easier, to prominently display the web link to use on the profile itself. You'll probably have to add this in /sites/effectivealtruismhub.com/themes/corporateclean/user-profile.tpl.php using the Drupal arg() function (or finding where Drupal stashes the URL or user ID of the profile, in which case you could `print "http://effectivealtruismhub.com/user/$USER_ID"`).

**Comment by tog22** [comment]
@cdc482 given the above do you think you should make this your top priority?

**Comment by cdc482** [comment]
I'm happy to make this my priority. To be clear, this should be my first
priority followed by the hiltons task suggester, right?

**Comment by tog22** [comment]
Yes

**Comment by cdc482** [comment]
I agree with your suggestion about using a drupal hook in template.php. Definitely seems like the easiest solution. I am considering a couple other options:
1. Adding a your profile/signout button to replace the sign up/login button in the search area block. (Visible to authenticated users etc.)
2. Using token's in the path for the Your Profile menu button (easier said than done of course). 

Anyway, it'd be nice to get some feedback if anyone has a couple minutes as to which of these are worth pursuing. Also, I've run into a problem modifying php files. For some reason, the changes are not reflected on the site, and I'm sure it's not an issue with cache. I'm sure I'll figure it out eventually, but I just thought I'd ask in case anyone has experienced this before.

**Comment by tog22** [comment]
(Note for others: I helped Chris sort out the cache issue, you need
to click clear caches on
http://diffdocroot.effectivealtruismhub.com/admin/config/development/performance
when that happens.)

## 

I receive a lot of email and limit my time spent on it, so may take a while
to see or reply to messages; I may be able to reply more quickly to emails
with an indication of timeframes (e.g. with 'urgent' in the subject line).

Contact information: http://bit.ly/tagoogleplus

**Comment by tog22** [comment]
which php files have you changed chris?

**Comment by tog22** [comment]
Option 1 is good thinking! That sounds so much quicker and easier that I'd
think it's worth doing for now and moving onto other priorities if you
agree. I imagine it should get just as many clicks and profile creations as
the old 'Your Profile' menu link would? Feel free to suggest tweaks to the
wording if you think it'd help with that.

**Comment by cdc482** [comment]
Glad you like option 1. Makes sense to me as well. I'll add a sign out button to the your profile page, and work on this option for now.

**Comment by tog22** [comment]
:+1:

**Comment by tog22** [comment]
Looks good, just theme it to look exactly like 'LOG IN OR SIGN UP'. Nice initiative adding 'sign out', you'd need to make that link to /user/logout.

**Comment by tog22** [comment]
@haryshwaran may be interested in taking this on if you can't do so soon @cdc482 - he may ask you for pointers!

**Comment by tog22** [comment]
@haryshwaran just added 'LOG OUT'. All that's needed to complete this issue is now to add a link alongside it to 'YOUR PROFILE' which points to the user's unique URL (e.g. /user/1 or /user/tom-ash ) rather than /user. @cdc482 may be able to explain his progress on this - I think it involved using Drupal tokens. If not I'd recommend googling those @haryshwaran .

**Comment by haryshwaran** [comment]
For I have added a "MY PROFILE" button and linked it to /user so that there is something available right now for users to click on and view their profile. Will look into drupal tokens and resolve this.

**Comment by tog22** [comment]
Ah, that would absolutely be a good idea, but unfortunately we have reason not to want it (briefly, because it'll lead people to share the wrong URL for their EA Profiles in a new EA Forum feature coming up). We actually had a link to /user before but took it down for this reason. Thanks for what would otherwise have been a totally sensible initiative, but could you reverse it if you've put it on prod?

**Comment by haryshwaran** [comment]
Ah, I see.. Got it, removed it. Will see how to about it using drupal tokens

**Comment by haryshwaran** [comment]
Link added via custom module. To do ( low priority ) will be to make the links themeable with drupal themes.

**Comment by tog22** [comment]
@haryshwaran low priority, but can you make 'my profile' say 'your profile' and split LOG IN OR SIGN UP into 2 links, to /user/login & /user/register . your choice how to style this.

**Comment by haryshwaran** [comment]
We will be changing the theme very soon in any case, we could club the changes along with the theme change. Could we close this issue?

**Comment by tog22** [comment]
We may as well leave it open as we'd want it in whatever theme we use. :smile:
